app/workers/migrate_data_to_work_field_worker.rb

```diff
@@ -8,7 +8,7 @@ def perform(profile_id)
     profile = Profile.find_by(id: profile_id)
     return unless profile
 
-    work_info = profile.employment_title
+    work_info = profile.employment_title.dup
     work_info << " at #{profile.employer_name}" if profile.employer_name.present?
```

Comment:
Agreed, `+=` seems clearer here than the `.dup` version. 

> I prefer to avoid mutation when feasible

Funnily enough, so do I. And one of the few times I don't it comes back to bite me. 

Anyway, I take some solace in this being somewhat difficult to spot, even if you know that this assignment was by reference and not by value, and only became a problem because it was a reference into the data structure that was later saved. Funnily enough, my first version was something like this:

```ruby
work_info = profile.employment_title
work_info = "#{work_info} at #{profile.employer_name}" if profile.employer_name.present?
```

That was before I decided to save on string allocation. Something something root of all evil 😉